clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -110,4 +114,17 @@ public String toString() {
             throw new SchemaException("Not an Object[].");
         }
     }
+
+    @Override
+    public String typeName() {
+        return ARRAY_TYPE_NAME;
+    }
+
+    @Override
+    public String documentation() {
+        return "Represents a sequence of objects of a given type T. " +
+                "Type T can be either a primitive type (e.g. " + STRING + ") or a structure. " +
+                "First, the length N is given as an " + INT32 + ". Then N instances of type T follow. " +
+                "In protocol documentation an array of T instances is referred to as [T].";
```

Comment:
Might want to make a note that a nullable array is represented with a length of -1 in the protocol.

Reply:
Thanks for the suggestion.